tests/model/model.cpp

```diff
@@ -552,3 +554,171 @@ TEST(Model, setAndCheckIdsAllEntities)
+        "  </component>\n"
+        "</model>\n";
+
+    const std::string in =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"myModel\">\n"
+        "  <units name=\"myUnitsThatIUse\">\n"
+        "    <unit units=\"second\"/>\n"
+        "  </units>\n"
+        "   <component name=\"myComponent\">\n"
+        "       <variable name=\"a\" units=\"second\"/>\n"
+        "       <math xmlns=\"http://www.w3.org/1998/Math/MathML\" xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\">\n"
+        "           <apply>"
+        "             <eq/>\n"
+        "             <ci>a</ci>\n"
+        "             <cn cellml:units=\"myUnitsThatIUse\">1</cn>\n"
+        "           </apply>\n"
+        "       </math>\n"
+        "   </component>\n"
+        "</model>";
+
+    // Create the model by parsing the string above.
+    auto parser = libcellml::Parser::create();
+    auto importedModel = parser->parseModel(in);
+
+    auto validator = libcellml::Validator::create();
+
+    // No problems with the imported model.
+    validator->validateModel(importedModel);
+    EXPECT_EQ(size_t(0), validator->errorCount());
+
+    // The model myModel already has myUnitsThatIUse defined.
+    auto model = libcellml::Model::create("myModel");
+    auto u = libcellml::Units::create("myUnitsThatIUse");
+    u->addUnit("metre");
+    model->addUnits(u);
+
+    auto c = libcellml::Component::create("c");
+
+    auto importSource = libcellml::ImportSource::create();
+    importSource->setUrl("not_required_resolving_import_manually");
+    importSource->setModel(importedModel);
+
+    c->setImportReference("myComponent");
+    c->setImportSource(importSource);
+    model->addComponent(c);
+
+    EXPECT_FALSE(model->hasUnresolvedImports());
+    model->flatten();
+
+    validator->validateModel(model);
+    EXPECT_EQ(size_t(0), validator->errorCount());
+
+    // I would expect that the name of the cn elements units would need
+    // to change as the importing model already has units of that name.
+    auto printer = libcellml::Printer::create();
+    EXPECT_EQ(e, printer->printModel(model));
+}
+
+TEST(Model, importUnitsDuplicated)
+{
+    const std::string in =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"myImportedModel\">\n"
+        "  <units name=\"millisecond\">\n"
+        "    <unit prefix=\"milli\" units=\"second\"/>\n"
+        "  </units>\n"
+        "  <units name=\"per_millisecond\">\n"
+        "    <unit exponent=\"-1\" units=\"millisecond\"/>\n"
+        "  </units>\n"
+        "  <component name=\"myComponent\">\n"
+        "    <variable name=\"alpha_n\" units=\"per_millisecond\" initial_value=\"1\"/>\n"
+        "    <variable name=\"beta_n\" units=\"per_millisecond\" initial_value=\"2\"/>\n"
+        "  </component>\n"
+        "</model>";
+
+    const std::string expectedModelString =
```

Comment:
Consistency: replace `expectedModelString` with `e`? Also, move it at the beginning of the test?

Reply:
Done (though see comment above)

Reply:
This is test code, so we have been using short names. Not my personal preference, but we ought to be consistent.